app/res/layout/menu_grid_item.xml

```diff
@@ -1,28 +1,55 @@
 <?xml version="1.0" encoding="utf-8"?>
 <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
-              android:layout_width="fill_parent"
-              android:layout_height="fill_parent"
-              android:textAppearance="@style/EntityItemText"
-              android:orientation="vertical">
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:textAppearance="@style/EntityItemText"
+    android:orientation="vertical">
 
-    <ImageView
-        android:id="@+id/row_img"
-        android:layout_gravity="center"
-        android:layout_width="@dimen/menu_grid_icon_size"
-        android:layout_height="@dimen/menu_grid_icon_size"
-        style="@style/ListContentV2"
-        />
+    <RelativeLayout
+        style="@style/GridViewContentWithBadge" >
+
+        <ImageView
+            android:id="@+id/row_img"
+            android:layout_width="@dimen/menu_grid_icon_size"
+            android:layout_height="@dimen/menu_grid_icon_size" />
+
+        <RelativeLayout
+            android:id="@+id/badge_view"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_alignParentBottom="true"
+            android:visibility="gone">
+
+            <ImageView
+                android:layout_width="@dimen/numeric_badge_width"
```

Comment:
this is a not-great solution to the fact that I can't size the badge ImageView dynamically based upon the badge's text, since the TextView isn't actually inside of the ImageView in the xml, it's just overlaid on top of it (all sorts of weird stuff started to happen when I tried to set it up that way - I think maybe you can't add elements within an ImageView?). So right now I'm just setting the `numeric_badge_width` dimension to be twice the `numeric_badge_font_size`, under the assumption that the text being shown will be 2 digits long or less... Very open to a better suggestion.